drainer/server.go

```diff
@@ -251,7 +251,7 @@ func (s *Server) Start() error {
 	}()
 
 	s.tg.GoNoPanic("collect", func() {
-		defer s.Close()
+		defer func() { go s.Close() }()
```

Comment:
it is ok now, but I think maybe it is better to define a new function like `Stop` to do some thing in `Close`, and only the main thread call the `Close` function to do `s.tg.Wait()`